src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorSpectrum.cs

```diff
@@ -0,0 +1,1784 @@
+				// of the third dimension, and one bitmap at the maximum.  Then we set the second's opacity at whatever the value of
+				// the third dimension is - e.g., a saturation of 0.5 implies an opacity of 50%.
+				// In the case where the third dimension is hue, we need six: one bitmap corresponding to red, yellow, green, cyan, blue, and purple.
+				// We'll then blend between whichever colors our hue exists between - e.g., an orange color would use red and yellow with an opacity of 50%.
+				// This optimization does incur slightly more startup time initially since we have to generate multiple bitmaps at once instead of only one,
+				// but the running time savings after that are *huge* when we can just set an opacity instead of generating a brand new bitmap.
+				if (shape == ColorSpectrumShape.Box)
+				{
+					for (int x = minDimensionInt - 1; x >= 0; --x)
+					{
+						for (int y = minDimensionInt - 1; y >= 0; --y)
+						{
+							if (workItem.Status == AsyncStatus.Canceled)
+							{
+								break;
+							}
+
+							FillPixelForBox(
+								x, y, hsv, minDimensionInt, components, minHue, maxHue, minSaturation, maxSaturation, minValue, maxValue,
+								bgraMinPixelData, bgraMiddle1PixelData, bgraMiddle2PixelData, bgraMiddle3PixelData, bgraMiddle4PixelData, bgraMaxPixelData,
+								newHsvValues);
+						}
+					}
+				}
+				else
+				{
+					for (int y = 0; y < minDimensionInt; ++y)
+					{
+						for (int x = 0; x < minDimensionInt; ++x)
+						{
+							if (workItem.Status == AsyncStatus.Canceled)
+							{
+								break;
+							}
+
+							FillPixelForRing(
+								x, y, minDimensionInt / 2.0, hsv, components, minHue, maxHue, minSaturation, maxSaturation, minValue, maxValue,
+								bgraMinPixelData, bgraMiddle1PixelData, bgraMiddle2PixelData, bgraMiddle3PixelData, bgraMiddle4PixelData, bgraMaxPixelData,
+								newHsvValues);
+						}
+					}
+				}
+			};
+
+			if (m_createImageBitmapAction != null)
+			{
+				m_createImageBitmapAction.Cancel();
+			}
+
+			m_createImageBitmapAction = ThreadPool.RunAsync(workItemHandler);
+			var strongThis = this;
+			m_createImageBitmapAction.Completed = new AsyncActionCompletedHandler(
+			(IAsyncAction asyncInfo, AsyncStatus asyncStatus) =>
+			{
+				if (asyncStatus != AsyncStatus.Completed)
+				{
+					return;
+				}
+
+				strongThis.m_createImageBitmapAction = null;
+
+				// Uno Doc: Assumed normal priority is acceptable
+				_ = strongThis.Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+				{
+					int pixelWidth = (int)Math.Round(minDimension);
+					int pixelHeight = (int)Math.Round(minDimension);
+
+					// Uno Doc: In C# (unlike C++) the existing 'components' var is captured by the lambda so it must be renamed here
+					ColorSpectrumComponents components2 = strongThis.Components;
+
+					// UNO TODO
+					//if (SharedHelpers.IsRS2OrHigher)
+					//{
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)